crates/matrix-sdk/src/room/timeline/mod.rs

```diff
@@ -443,6 +347,136 @@ impl Timeline {
     }
 }
 
+/// Builder that allows creating and configuring various parts of a
+/// [`Timeline`].
+#[must_use]
+#[derive(Debug)]
+pub(crate) struct TimelineBuilder {
```

Comment:
It's not huge right now, but I would prefer having this in a separate `builder` module since this file is already above 500 lines long, and this is easy enough to move out.